av1an-cli/src/lib.rs

```diff
@@ -442,164 +442,217 @@ fn confirm(prompt: &str) -> io::Result<bool> {
   }
 }
 
-pub fn parse_cli(args: CliOpts) -> anyhow::Result<EncodeArgs> {
-  let temp = if let Some(path) = args.temp.as_ref() {
-    path.to_str().unwrap().to_owned()
+/// Given Folder and File path as inputs
+/// Converts them all to file paths
+/// Converting only depth 1 of Folder paths
+pub fn convert_input(path: &Path) -> anyhow::Result<Vec<PathBuf>> {
+  // TODO: to validate file extensions
+  // let valid_media_extensions = ["mkv", "mov", "mp4", "webm", "avi", "qt", "ts", "m2t", "py", "vpy"];
+
+  if path.is_file() {
+    Ok(vec![path.to_path_buf()])
+  } else if path.is_dir() {
+    Ok({
+      let dir = std::fs::read_dir(path)?;
+      dir
+        .into_iter()
+        .filter_map(Result::ok)
+        .filter_map(|d| {
+          if let Ok(file_type) = d.file_type() {
+            if file_type.is_file() {
+              Some(d.path())
+            } else {
+              None
+            }
+          } else {
+            None
+          }
+        })
+        .collect()
+    })
   } else {
-    format!(".{}", hash_path(args.input.as_path()))
-  };
+    Ok(vec![])
```

Comment:
This should probably return an error instead of an empty vector, instead of relying on the caller to check for the empty case.